diff --git a/exercises/quiz3.rs b/exercises/quiz3.rs
index 15dc469..a1b673b 100644
--- a/exercises/quiz3.rs
+++ b/exercises/quiz3.rs
@@ -14,7 +14,6 @@
 
 // Execute `rustlings hint quiz3` or use the `hint` watch subcommand for a hint.
 
-// I AM NOT DONE
 
 pub struct ReportCard {
     pub grade: f32,
@@ -28,6 +27,23 @@ impl ReportCard {
             &self.student_name, &self.student_age, &self.grade)
     }
 }
+impl std::fmt::Debug for ReportCard {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{} ({}) - achieved a grade of {}",
+            &self.student_name, &self.student_age, &self.grade)
+    }
+}
+impl std::fmt::Display for ReportCard {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let score = match self.grade {
+            g if g > 5.0 => "A+",
+            g if g > 4.0 => "B",
+            _ => "F-",
+        };
+        write!(f, "{} ({}) - achieved a grade of {}",
+            &self.student_name, &self.student_age, score)
+    }
+}
 
 #[cfg(test)]
 mod tests {
@@ -41,7 +57,7 @@ mod tests {
             student_age: 12,
         };
         assert_eq!(
-            report_card.print(),
+            format!("{:?}", report_card),
             "Tom Wriggle (12) - achieved a grade of 2.1"
         );
     }
@@ -50,12 +66,12 @@ mod tests {
     fn generate_alphabetic_report_card() {
         // TODO: Make sure to change the grade here after you finish the exercise.
         let report_card = ReportCard {
-            grade: 2.1,
+            grade: 5.1,
             student_name: "Gary Plotter".to_string(),
             student_age: 11,
         };
         assert_eq!(
-            report_card.print(),
+            format!("{}", report_card),
             "Gary Plotter (11) - achieved a grade of A+"
         );
     }
